Expensify/App
# Highlight all errors together in BankAccount step

cc @marcaaron

Brought some of the functionality implemented in this [dead PR](https://github.com/Expensify/App/pull/4890)

### Details

Changes: 

- Added missing inline error for field `Account Number`
- "Save & continue" button is only disabled if there is some field empty or the checkbox is unchecked
- Support highlighting multiple errors at the same time
- Clear error of field being edited

These changes apply only to BankAccount step in VBA flow.

### Fixed Issues

$ https://github.com/Expensify/App/issues/4785

### Tests AND QA Steps

1. Start with a non-public domain account that has not yet set up a workspace or bank account
2. Create a new workspace by tapping on the green FAB button
3. Navigate to `/bank-account` by selecting the workspace and tapping the "Get Started" button
4. Tap button to add account manually: `Connect manually`
5. **_Expected result_**: the "Save & continue" button should be disabled.
6. Enter an invalid `Routing number`, i.e.: `1` (front-end validation)
7. **_Expected result_**: the "Save & continue" button should remain disabled.
6. Enter an invalid `Account number`, i.e.: `1` (front-end validation)
8. Click "Save & continue"
9. **_Expected result_**: You should get the error modal with the generic message: `Please double check any highlighted fields and try again.` and see both fields with inline highlighted errors
![Screen Shot 2021-09-02 at 5 06 30 PM](https://user-images.githubusercontent.com/87341702/131931118-257f3edc-f839-48d4-9f39-1ae93274a481.png)
10. Focus on any of the inputs, type something more
11. **_Expected result_**: the inline error and highlight should disappear only for that input.
![Screen Shot 2021-09-02 at 5 07 15 PM](https://user-images.githubusercontent.com/87341702/131931159-0e369057-b7eb-4796-8686-84ed4da966f1.png)
12. Enter `33333332234` as `Routing number` and `111` as the `Account number`
13. Click "Save & continue"
14. **_Expected result_**: We should see the loading transition (data is sent to back-end), and then we should see the form again with an inline error for `Routing number` and the error modal with the generic message.

https://user-images.githubusercontent.com/87341702/131933996-47f11549-81ca-4c41-9003-e7e1ce41b724.mov





### Tested On


### Screenshots


#### Web


#### Mobile Web


#### Desktop


#### iOS


#### Android
